src/software/simulated_tests/kickoff_friendly_play_test.cpp

```diff
@@ -0,0 +1,73 @@
+#include "software/ai/hl/stp/play/kickoff_friendly_play.h"
+
+#include <gtest/gtest.h>
+
+#include "software/simulated_tests/simulated_test_fixture.h"
+#include "software/simulated_tests/validation/validation_function.h"
+#include "software/test_util/test_util.h"
+#include "software/time/duration.h"
+#include "software/time/timestamp.h"
+#include "software/world/world.h"
+
+class KickoffFriendlyPlayTest : public SimulatedTest
+{
+};
+
+TEST_F(KickoffFriendlyPlayTest, test_kickoff_friendly_play)
+{
+    World world = ::Test::TestUtil::createBlankTestingWorld();
+
+    world = ::Test::TestUtil::setEnemyRobotPositions(
+        world,
+        {
+            Point(0.6, 0),
+            Point(0.2, 2.5),
+            Point(0.2, -2.5),
+            world.field().enemyGoalCenter(),
+            world.field().enemyDefenseArea().negXNegYCorner(),
+            world.field().enemyDefenseArea().negXPosYCorner(),
+        },
+        Timestamp::fromSeconds(0));
+
+    world = ::Test::TestUtil::setFriendlyRobotPositions(world,
+                                                        {
+                                                            Point(-3, 2.5),
+                                                            Point(-3, 1.5),
+                                                            Point(-3, 0.5),
+                                                            Point(-3, -0.5),
+                                                            Point(-3, -1.5),
+                                                            Point(-3, -2.5),
+                                                        },
+                                                        Timestamp::fromSeconds(0));
+    world.mutableFriendlyTeam().assignGoalie(0);
+
+    world.mutableBall() = Ball(Point(0, 0), Vector(0, 0), Timestamp::fromSeconds(0));
+
+    std::vector<ValidationFunction> validation_functions = {
+        // This will keep the test running for 9.5 seconds to give everything enough
+        // time to settle into position and be observed with the Visualizer
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
+            {
+                yield();
+            }
```

Comment:
I guess this is the bit that would be modified to test conditions? Would be good to add a TODO with an issue# here

Reply:
done